File: FlexServ Backend/SpringBoot/TransactionService/src/main/java/com/flexserv/dto/request/BookingCreateRequest.java
package com.flexserv.dto.request;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalTime;

import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class BookingCreateRequest {

    @NotNull(message = "Customer ID is required")
    private Long customerId;

    @NotNull(message = "Provider ID is required")
    private Long providerId;

    @NotNull(message = "Service ID is required")
    private Long serviceId;

    @NotNull(message = "Address ID is required")
    private Long addressId;

    @NotNull(message = "Booking date is required")
    private LocalDate bookingDate;

    @NotNull(message = "Booking time is required")
    private LocalTime bookingTime;

    private BigDecimal totalPrice;
}

File: FlexServ Backend/DotNet/TransactionService/Dtos/BookingCreateRequest.cs
using System.ComponentModel.DataAnnotations;

namespace TransactionService.Dtos;

public class BookingCreateRequest
{
    [Required(ErrorMessage = "Customer ID is required")]
    public long CustomerId { get; set; }

    [Required(ErrorMessage = "Provider ID is required")]
    public long ProviderId { get; set; }

    [Required(ErrorMessage = "Service ID is required")]
    public long ServiceId { get; set; }

    [Required(ErrorMessage = "Address ID is required")]
    public long AddressId { get; set; }

    [Required(ErrorMessage = "Booking date is required")]
    public DateOnly BookingDate { get; set; }

    [Required(ErrorMessage = "Booking time is required")]
    public TimeOnly BookingTime { get; set; }

    public decimal? TotalPrice { get; set; }
}
